source/opt/ir_context.h

```diff
@@ -204,6 +205,24 @@ class IRContext {
     return decoration_mgr_.get();
   };
 
+  // Returns a pointer to the constant manager.  If no constant manager has been
+  // created yet, it creates one.  NOTE: Once created, the constant manager
+  // remains active and it is never re-built.
+  opt::analysis::ConstantManager* get_constant_mgr() {
+    if (!constant_mgr_)
+      constant_mgr_.reset(new opt::analysis::ConstantManager(this));
+    return constant_mgr_.get();
+  }
+
+  // Returns a pointer to the type manager.  If no type manager has been created
+  // yet, it creates one. NOTE: Once created, the type manager remains active it
```

Comment:
I don't think this is appropriate currently. The TypeManager currently cannot add types after its initial analysis. So if a new type is ever added, it will not show up in the TypeManager. I have filed #1071 to improve the TypeManager.

Reply:
I've made TypeManager::AnalyzeTypes a public method for now.  I want to keep this re-factor as NFC as possible.  I'd rather extend the TypeManager as part of the fix for #1071.